tests/BCL-Tests/Xamarin.Android.Bcl-Tests/Xamarin.Android.Bcl-Tests.targets

```diff
@@ -5,71 +5,90 @@
-  </ItemGroup>
-  <Target Name="_CreateTestResources"
-      BeforeTargets="ResolveAssemblyReferences"
-      DependsOnTargets="_ExtractTestResources"
-      Inputs="@(_TestResource)"
-      Outputs="$(IntermediateOutputPath)%(_TestResource.Culture)\%(_TestResource.Name)">
-    <MakeDir Directories="$(IntermediateOutputPath)%(_TestResource.Culture)" />
-    <AL
-        OutputAssembly="$(IntermediateOutputPath)%(Culture)\%(Name)"
-        Version="1.0.0.0"
-        Culture="%(Culture)"
-        EmbedResources="@(_TestResource)"
+    <Copy
+        Condition="Exists('..\..\..\bin\$(Configuration)\bcl-tests\%(_TestAssembly.RecursiveDir)%(_TestAssembly.Filename).pdb')"
+        SourceFiles="@(_TestAssembly->'..\..\..\bin\$(Configuration)\bcl-tests\%(RecursiveDir)%(Filename).pdb')"
+        DestinationFiles="@(_TestAssembly->'$(IntermediateOutputPath)%(RecursiveDir)%(Filename).pdb')"
     />
   </Target>
-  <Target Name="_ExtractTestResources"
-      Inputs="..\..\..\bin\$(Configuration)\bcl-tests\bcl-tests.zip"
-      Outputs="@(_TestResource)">
-    <SystemUnzip
-        SourceFiles="..\..\..\bin\$(Configuration)\bcl-tests\bcl-tests.zip"
-        HostOS="$(HostOS)"
-        DestinationFolder="..\..\..\bin\$(Configuration)\bcl-tests"
+  <Target Name="_WriteAssembliesList"
+      Outputs="nunit-assemblies.txt;xunit-assemblies.txt"
+      BeforeTargets="CoreCompile">
+    <ItemGroup>
+      <_NUnitTestAssembly Include="monodroid_corlib_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_I18N.CJK_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_I18N.MidEast_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_I18N.West_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_Mono.CSharp_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_Mono.Data.Sqlite_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_Mono.Data.Tds_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_Mono.Security_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Core_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Data_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.IO.Compression.FileSystem_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.IO.Compression_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Json_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Net.Http_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Numerics_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Runtime.Serialization_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.ServiceModel.Web_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Transactions_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Xml_test.dll" />
+      <_NUnitTestAssembly Include="monodroid_System.Xml.Linq_test.dll" />
+    </ItemGroup>
```

Comment:
Why not use the previously generated `_TestAssembly` items with the `nunit` `TestType` here?

Reply:
The trick here is `_TestAssembly` will be *all* test assemblies generated for *all* the `monodroid` profile BCL assemblies, while `_NUnitTestAssembly` and `_XUnitTestAssembly` are the test assemblies we _know_ are working and should not be failing.

`_NUnitTestAssembly ` and `_XUnitTestAssembly` and then a subset of `_TestAssembly`, and we should work on getting them to not be a subset anymore.

Reply:
You can create `_NunitTestAssembly` and `_XunitTestAssembly` by batching over the `_TestAssembly` items and matching their `TestType`, or you can create a custom task which will take all the `_TestAssembly` items as input and output appropriate item groups. The point is to have a single source set of assemblies, to avoid confusion and potential errors.

Reply:
I did just what you described.